app/assets/javascripts/controllers/topology/network_topology_controller.js

```diff
@@ -183,16 +183,16 @@ function NetworkTopologyCtrl($scope, $http, $interval, $location, topologyServic
   function getNetworkTopologyData(response) {
     var data = response.data;
 
-    var currentSelectedKinds = $scope.kinds;
-    $scope.items = data.data.items;
-    $scope.relations = data.data.relations;
-    $scope.kinds = data.data.kinds;
+    var currentSelectedKinds = vm.kinds;
+    vm.items = data.data.items;
+    vm.relations = data.data.relations;
+    vm.kinds = data.data.kinds;
```

Comment:
`vm.kinds = data.data.kinds` will not work with, because legged buttons are looking for kinds in scope.
so there must be something like `vm.kinds = $scope.kinds = data.data.kinds`, until we make our own legend component.